### PR TITLE
Add colony export command

### PR DESCRIPTION
**Pull request by someaddons**
# Changes proposed in this pull request:
Adds a colony export command, which exports a small debug world folder of the colony, its data and its region files. It is intended to allow easier debugging of certain worlds, esp when looking at a certain colony on a big server world.

Includes these folders of the original world, region is limited to only chunks of a certain colony
image

Review please

### CODE REVIEW

**Comment by Raycoms** [comment]
:shipit:

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 1]
```diff
@@ -408,6 +408,7 @@
   "com.minecolonies.command.raidnow.success": "Raid started for colony %s, take care!",
   "com.minecolonies.command.raidtonight.success": "A raid on colony %s is coming tonight!",
   "com.minecolonies.command.loadbackup.success": "Successfully loaded backup.",
+  "com.minecolonies.command.export.success": "Exported colony to zip: %s",
```
Comment (ravenbuilder934) [inline src/main/resources/assets/minecolonies/lang/en_us.json:411]: and if it's *not* a success?

**Comment by Mekle001** [comment]
Can we include a list of mods and their versions in the export?

File: src/main/resources/assets/minecolonies/lang/en_us.json (hunk 1 above)
Comment (someaddons) [reply src/main/resources/assets/minecolonies/lang/en_us.json:411]: then the %s has the error message

**Comment by someaddons** [comment]
hm for which usecase do you want a list of mods added? the data gathering here pretty much includes no mod data

**Comment by uecasm** [comment]
List of mods would be useful for the "oh, you're using mod X, that causes bug Y" cases.

**Comment by someaddons** [comment]
but that would normally be before asking the player to provide a world no?

**Comment by uecasm** [comment]
If they remember to tell you, or if they've previously shown logs, sure.  But that's probably not always the case, and more info never hurts, especially if it's easy to capture like the modlist should be.  (Including recent logs automatically could be useful too, but that's probably harder after the fact unless you can just copy the existing logfile based on known relative path.)

**Comment by someaddons** [comment]
logs and mod list probably would be done the same way via paths n filenames, the loaded mods dont have the filename version info

**Comment by marchermans** [comment]
There is a way to get that via the ModList.

**Comment by someaddons** [comment]
Ye the modlist was the first thing I checked, it doesnt have actual filenames just mod.toml versions, so only mods who provided one would show. Just doing it via files now as its similar to the latestlog anyways

File: src/main/java/com/minecolonies/coremod/util/BackUpHelper.java [hunk 2]
```diff
@@ -382,4 +409,164 @@ public static void loadColonyBackup(final int colonyId, final RegistryKey<World>
...
+    /**
+     * Exports a certain colony and its part of the world and colony data data to a zip
+     *
+     * @param colony colony to export
+     * @return file Path and name
+     */
+    public static String exportColony(final IColony colony)
+    {
+        final MinecraftServer server = ServerLifecycleHooks.getCurrentServer();
+        final File topworldDir = server.func_240776_a_(FolderName.DOT).toFile();
```
Comment (Nightenom) [inline src/main/java/com/minecolonies/coremod/util/BackUpHelper.java:422]: just side comment, using nio Path would be appreciated as it have things like toAbsolutePath, resolve, normalize and relativize which would remove the replace (and File.separator) hacks below

Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/util/BackUpHelper.java:422]: yea not worth the effort, would have to do it in the whole file as we use it in other places too

**Pull request merged**